src/sentry/search/eap/resolver.py

```diff
@@ -971,3 +977,85 @@ def resolve_function(self, column: str, match: Match | None = None) -> tuple[
         resolved_context = None
         self._resolved_function_cache[column] = (resolved_function, resolved_context)
         return self._resolved_function_cache[column]
+
+    def resolve_equations(self, equations: list[str]) -> tuple[
+        list[ResolvedEquation],
+        list[VirtualColumnDefinition],
+    ]:
+        formulas = []
+        contexts = []
+        for equation in equations:
+            formula, context = self.resolve_equation(equation)
+            formulas.append(formula)
+            contexts.extend(context)
+        return formulas, contexts
+
+    def resolve_equation(self, equation: str) -> tuple[
+        ResolvedEquation,
+        list[VirtualColumnDefinition],
+    ]:
+        """Resolve an equation creating a ResolvedEquation object, we don't just return a Column.BinaryFormula since
+        it'll help callers with extra information, like the existence of aggregates and the search type
+        """
+        operation, fields, functions = arithmetic.parse_arithmetic(equation)
+        lhs, lhs_contexts = self._resolve_operation(operation.lhs) if operation.lhs else (None, [])
+        rhs, rhs_contexts = self._resolve_operation(operation.rhs) if operation.rhs else (None, [])
+        return (
+            ResolvedEquation(
+                public_alias=f"equation|{equation}",
+                # Type of equations can become complex very quickly. We could try to make sure all the columns have the
+                # same type and return that, but then ratios would have types eg. (p75/p50), as well managing multiple
+                # column types is strange too (p75+count). Keeping this as just `number` for now
+                search_type="number",
+                operator=constants.ARITHMETIC_OPERATOR_MAP[operation.operator],
+                lhs=lhs,
+                rhs=rhs,
+                is_aggregate=len(functions) > 0,
```

Comment:
Hmm, technically not every function is an aggregate, (for example if you just divide one column by another). Although i think in our case this might not be a problem

Reply:
Sorry what do you mean?

Reply:
I think ```
is_aggregate:len(functions)>0
```
Would technically be incorrect if we had a function that was like ```
my_function()=attributeA/attributeB
```

Because my_function is not an aggregate function.

Reply:
Iirc, we store an is_aggregate property on function definitions, so technically that would be the best source of truth

Reply:
Yeah good point, i'll have to resolve all the functions to check that